shared/naturalcrit/codeEditor/fold-code.js

```diff
@@ -0,0 +1,28 @@
+module.exports = {
+	registerHomebreweryHelper : function(CodeMirror) {
+		CodeMirror.registerHelper('fold', 'homebrewery', function(cm, start) {
+			const matcher = /^\\page.*/;
+			const prevLine = cm.getLine(start.line - 1);
+
+			if(start.line === cm.firstLine() || prevLine.match(matcher)) {
+				const lastLineNo = cm.lastLine();
+				let end = start.line, nextLine = cm.getLine(start.line + 1);
+
+				while (end < lastLineNo) {
+					if(nextLine.match(matcher)) {
+						return {
+							from : CodeMirror.Pos(start.line, 0),
+							to   : CodeMirror.Pos(end, cm.getLine(end).length)
+						};
+					}
+					++end;
+					nextLine = cm.getLine(end + 1);
+				}
+
+				return null;
```

Comment:
At this point, if nothing has been returned yet from the while loop, by definition we *must* be at the end of the document, right? So can we just return everything from the start point to the end?

```suggestion
				return {
					from : CodeMirror.Pos(start.line, 0),
					to   : CodeMirror.Pos(end, cm.getLine(end).length)
				};
```

Reply:
That would allow the section from the last `\page` to the end of the document to be folded, however while I was writing this method I wondered if that might be a pointless area to fold. I was thinking that section is probably where most people will be doing work on a document, and thus folding it would not provide any benefit. 

If we think it makes more sense for the functionality to be as consistent as possible then yes by all means we should add that.

Reply:
Yes, I think we should make it consistent that every page is foldable.

1) We can't assume how and where users want to fold. Users might want to go back and edit somewhere in the middle of the document and close up everything else, etc.

2) If it works everywhere except one spot, users will assume it's a bug.

Reply:
Added that!